slides/testing/validator/duplicate-id-validator.ts

```diff
@@ -0,0 +1,61 @@
+import { FileParser } from './../parser/file-parser';
+import { IValidator } from './validator.interface';
+import { File } from './../parser/file';
+import * as fs from 'fs';
+import * as path from 'path';
+
+export class DuplicateIdValidator implements IValidator {
+
+  private duplicateSlides: any;
+  private basePath = path.resolve(__dirname, '../../../src/content/');
+  
+  get duplicates() {
+    return this.duplicateSlides;
+  }
+
+  validate(): boolean {
+    let files = this.getFiles();
+    let slideIds = this.getSlideIds(files);
+    this.duplicateSlides  = this.getDuplicates(slideIds);
+    return this.duplicateSlides.size === 0 ? true : false;
+  }
```

Comment:
@gvwilson I had a discussion with @rpdasilva about this. We are going to keep it as is, and use the return value of the validate method.